src/function_manager.jl

```diff
@@ -55,19 +55,26 @@ end
 # so "jlfun" seems reasonable
 const FUNCTION_MANAGER_NAMESPACE = "jlfun"
 
-Base.@kwdef struct FunctionManager
+Base.@kwdef mutable struct FunctionManager
     gcs_client::ray_jll.JuliaGcsClient
     functions::Dict{String,Any}
+
+    function FunctionManager(gcs_client, functions)
+        status = ray_jll.Connect(gcs_client)
+        ray_jll.ok(status) || error("Could not connect to GCS")
```

Comment:
a bit opinionated maybe? I figured with the `finalizer` calling `Disconnect` we were implicitly tying the `Connect` to the constructor?

Reply:
I think this is a step too far as now it's impossible to create a `FunctionManager` without connecting it to the GCS client. I can understand your concern with running `Disconnect` automatically on every `FunctionManager`. What if you just use `finalizer` on a single instance created in `_init_global_function_manager`?

Reply:
ah, so I didn't realise that was possible. the docs kinda suggest that it should be called on instantiatoin but perhaps that was my misunderstanding

Reply:
I'll rever this change and add the `finalizer` as you suggest